packages/launchpad/src/use-tracking/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect, useMemo } from 'react';
+import type { Task } from '../types';
+import type { SiteDetails } from '@automattic/data-stores';
+
+interface LogParams {
+	tasks: Task[];
+	checklistSlug: string | null;
+	context: string;
+	site: SiteDetails | null | undefined;
+}
+
+const stringifyTasks = ( tasks: Task[] ) => {
+	const prefix = ',';
+	const suffix = ',';
+	const separator = ',';
+
+	const taskIds = tasks.map( ( task ) => task.id ).join( separator );
+	return `${ prefix }${ taskIds }${ suffix }`;
+};
+
+export const useTracking = ( params: LogParams ) => {
+	const { tasks, checklistSlug, context, site } = params;
+	const completedSteps = useMemo( () => tasks.filter( ( task ) => task.completed ), [ tasks ] );
+	const taskNames = useMemo( () => stringifyTasks( tasks ), [ tasks ] );
+	const numberOfSteps = tasks.length;
+	const numberOfCompletedSteps = completedSteps.length;
+	const isCheckListCompleted = completedSteps.length === tasks.length;
+	const hasSite = !! site;
+
+	useEffect( () => {
+		if ( tasks.length === 0 || ! hasSite ) {
```

Comment:
Minor thought: it may be worth commenting on why we're doing skipping the effect for these two conditions.
```suggestion
		// If we don't have any tasks, or if we're still waiting for the site to load, don't trigger any tracking yet.
		if ( tasks.length === 0 || ! hasSite ) {
```

Reply:
I created a new flag to reduce the # of useEffect cans and explain better what we are doing.  I also added the comment as you suggested. 

0743aac2903a1b5240be4d45ea7cc9623396dac5